src/EditorFeatures/CSharpTest/BlockCommentCompletion/BlockCommentCompletionTests.cs

```diff
@@ -0,0 +1,346 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Commands;
+using Microsoft.CodeAnalysis.Editor.CSharp.BlockCommentCompletion;
+using Microsoft.CodeAnalysis.Editor.UnitTests.BlockCommentCompletion;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Utilities;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.BlockCommentCompletion
+{
+    public class BlockCommentCompletionTests : AbstractBlockCommentCompletionTests
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnStartLine0()
+        {
+            var code = @"
+    /*$$
+";
+            var expected = @"
+    /*
+     * $$
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnStartLine1()
+        {
+            var code = @"
+    /*$$*/
+";
+            var expected = @"
+    /*
+     $$*/
```

Comment:
VS2013 behavior. It's actually easier to insert a `*`, no need my buggy `BlockCommentEndsRightAfterCaret` then.